diesel/src/macros/mod.rs

```diff
@@ -551,6 +551,19 @@ macro_rules! print_sql {
     };
 }
 
+#[macro_export]
+/// This macro can only be used in combination with the `diesel_codegen` or
+/// `diesel_codegen_syntex` crates. It will not work on its own.
+///
+/// FIXME: Oh look we have a place to actually document this now.
+macro_rules! infer_schema {
+    ($database_url: expr) => {
+        #[derive(InferSchema)]
+        #[options(database_url=$database_url)]
+        struct _Dummy;
```

Comment:
I wonder if there's any better name here, even though it is just an implementation detail.